### ISSUE TITLE
Conda env creation very slow over NFS

### ISSUE THREAD

**Issue by mvdbeek** [galaxyproject/galaxy#3101]
Our conda installation is on a NFS filesystem, and activating environments there (with default or `conda_copy_dependencies = True`) is quite slow. Linking in bioperl takes around 3-5 minutes.
I suppose that's because conda isn't linking all files with hardlinks, though I'm a bit confused by that matter.

This is the output of ls -li

```
5485165401 -rwxrwxr-x 3 nobody 4294967294    39243 Jun 30  2015 xmlcatalog
5516844377 -rwxr-xr-x 1 nobody 4294967294    13695 Oct 27 10:37 xml_grep
5485165402 -rwxrwxr-x 3 nobody 4294967294   177418 Jun 30  2015 xmllint
5516844378 -rwxr-xr-x 1 nobody 4294967294     4035 Oct 27 10:37 xml_merge
5516844379 -rwxr-xr-x 1 nobody 4294967294     8060 Oct 27 10:37 xml_pp
5516844380 -rwxr-xr-x 1 nobody 4294967294     6551 Oct 27 10:37 xml_spellcheck
5516844381 -rwxr-xr-x 1 nobody 4294967294    21692 Oct 27 10:37 xml_split
5486343814 -rwxr-xr-x 3 nobody 4294967294    68638 Jan 12  2016 xmlwf
5516307736 -rwxr-xr-x 1 nobody 4294967294     1590 Oct 27 10:37 xpath
5516513158 -rwxr-xr-x 1 nobody 4294967294     2596 Oct 27 10:36 xslt-config
5485952547 -rwxr-xr-x 3 nobody 4294967294    65956 Feb  8  2013 xsltproc
5515656982 -rwxr-xr-x 1 nobody 4294967294     5218 Oct 27 10:36 xsubpp
5486343971 -rwxr-xr-x 3 nobody 4294967294   222105 Apr  2  2016 xz
5515656983 -rwxr-xr-x 1 nobody 4294967294    48551 Oct 27 10:36 zipdetails
```

So it seems that some files get hardlinked and other not, based on the reference count.
I'm not really sure of the underlying NFS architecture, but I believe that the NFS mount is actually distributed across multiple filesystems.

To work around the issue it would be cool if we could generate permanent (or cached ...) environments similar to how @jmchilton has implemented the mulled dependency resolution, i.e generate a hash of the combination of all individual tool requirements.

**Comment by lecorguille** [comment]
👍

**Comment by mvdbeek** [comment]
This has been solved to the extent possible with #3391.
